src/Store/Reducers/favoriteReducer.ts

```diff
@@ -0,0 +1,38 @@
+import {ForecastData} from '../../Components/weatherData.type';
+import {ActionType} from '../actions';
+
+const initialState: WeatherState = {favoriteForecast: []};
+
+export interface WeatherState {
+  favoriteForecast: ForecastData[];
+}
+
+export function favoriteReducer(
+  state: WeatherState = initialState,
+  action: ActionType,
+): WeatherState {
+  let newState;
+  switch (action.type) {
+    case 'TOGGLE_FAVORITE':
```

Comment:
avoid logic in your reducer, it should be 2 actions, one for favorite, one for unfavorite